pkg/registry/discovery/endpointslice/strategy.go

```diff
@@ -149,26 +149,38 @@ func dropDisabledFieldsOnCreate(endpointSlice *discovery.EndpointSlice) {
 				endpointSlice.Endpoints[i].Hints = nil
```

Comment:
The `if` line above this is redundant - can you clean it up while you are in here?

Reply:
cleaned up and synced `dropDisabledFieldsOnCreate` and `dropDisabledFieldsOnUpdate`.